src/dss/main.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+import click
+
+from dss.initialize import initialize
+from dss.logger import setup_logger
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+# Name for the environment variable storing kubeconfig
+KUBECONFIG_ENV_VAR = "DSS_KUBECONFIG"
+
+
+@click.group()
+def main():
+    """Command line interface for managing the DSS application."""
+
+
+@main.command(name="initialize")
+@click.option(
+    "--kubeconfig",
+    help="Specify the kubeconfig content. Defaults to the value of file at path on the KUBECONFIG environment variable.",  # noqa E501
+)
+def initialize_command(kubeconfig: str) -> None:
+    """
+    Initialize the Kubernetes cluster and deploy charms.
+    """
+    logger.info("Executing initialize command")
+
+    # If kubeconfig is provided, set it as an environment variable
+    if kubeconfig:
+        os.environ[KUBECONFIG_ENV_VAR] = kubeconfig
```

Comment:
Or if the user wants to execute something from a second terminal

Reply:
I believe we need to rethink the environment variable as storage.  This sounds like it only works for a subset of where we need it to. 

Can we just cache the file somewhere?  As a standalone python package I'm not sure where we'd put the file, but as a snap (which this will be deployed as) I think we have somewhere we can store files right?  Maybe we just write the config that is provided to that location?

Reply:
Agree with writing it to a file. This is also the only way other commands in the future can use this context to interact with the cluster.

I'd suggest the following:
1. Let's use an env var for the path we store this file (allows for configuration for dev env vs snap env)
    1. When the code is part of a snap maybe this will be `$HOME/snap/dss/common` https://ubuntu.com/robotics/docs/snap-data-and-file-storage
    2. For dev purposes, it could be a place here in the repo?
2. Later on once we'll look into the snapcraft structure we could look into a way to change this env var when calling the CLI https://github.com/canonical/data-science-stack/issues/25

@ca-scribner let me know what you think. And if we then decide to go with what proposed in 2 let's then add a comment in that issue to not forget we'll need this

Reply:
Yeah I was thinking something similar.  Its a bit tricky though since our source is `microk8s config` rather than an existing file.  `kubectl` does something a bit different as it accepts a path to a config file (and doesn't cache that path).  I wonder if we need two separate things:
* an environment variable that defines where we look for a kubeconfig, which by default could be the place we cache a previously provided kubeconfig
* a way to give the dss tool a kubeconfig file to cache (either as a flag to initialize, or as a separate command)

I'll play with this a bit then we can discuss

Reply:
Refactored to get something using file paths rather than content.  wdyt?  We can rework the defaults when we know more about how snaps can handle this

Reply:
Sounds good, added some suggestions for next steps
https://github.com/canonical/data-science-stack/pull/29#issuecomment-1968863944